sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileAsyncClient.java

```diff
@@ -848,25 +848,26 @@ private void channelCleanUp(AsynchronousFileChannel channel) {
      *
      * <p>List all ranges within the file range from 1KB to 2KB.</p>
      *
-     * {@codesnippet com.azure.storage.file.fileAsyncClient.listRanges#filerange}
+     * {@codesnippet com.azure.storage.file.fileAsyncClient.listRanges#filerange-duration}
      *
      * <p>For more information, see the
      * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/list-ranges">Azure Docs</a>.</p>
      *
      * @param range Optional byte range which returns file data only from the specified range.
+     * @param timeout An optional timeout to be applied to the network asynchronous operations.
      * @return {@link FileRange ranges} in the files that satisfy the requirements
      */
-    public PagedFlux<FileRange> listRanges(FileRange range) {
+    public PagedFlux<FileRange> listRanges(FileRange range, Duration timeout) {
```

Comment:
Same as the other async client, need package private methods that take the duration. The public API shouldn't have the parameter.